test/generate-async.test.ts

```diff
@@ -46,3 +46,81 @@ describe('generate-async', () => {
     expect(order).eql([1, 2])
   })
 })
+
+describe('firing-events', () => {
+  test('config-event', async () => {
+    const order: number[] = []
+    const uno = createGenerator()
+
+    uno.events.on('config', () => order.push(order.length))
```

Comment:
I am not very sure what `order.length` is used for, maybe we could just use a fixed number?